setProxyUrl and ETIMEOUT

### Category
- [ ] Enhancement
- [ ] Bug
- [X] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [   2.0.13   ]

Please specify what version(s) of SharePoint you are targeting: [    SPO     ]

### Expected / Desired Behavior / Question
The servers for upper environments (TST, UAT, PRD) are under corporate proxy so when I deploy there I need to set up the proxy configuration.
I used `setProxyUrl()` and it seems it works partially.
From the proxy logs we can see that it hits the SPO authentication URL, but after that the application seems to try to hit another Microsoft owned IP address and that call does not use the proxy settings - we cannot see it in the proxy logs.
The result is:
```
request to https://accounts.accesscontrol.windows.net/xxx-yyy/tokens/OAuth/2 failed, reason: connect ETIMEDOUT 40.126.7.66:443 stack:FetchError: request to https://accounts.accesscontrol.windows.net/xxx-yyy/tokens/OAuth/2 failed, reason: connect ETIMEDOUT 40.126.7.66:443
    at ClientRequest.<anonymous> (E:\inetpub\wwwroot\onestop-attachments\node_modules\node-fetch\lib\index.js:1345:11)
    at ClientRequest.emit (events.js:198:13)
    at TLSSocket.socketErrorListener (_http_client.js:392:9)
    at TLSSocket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```
NOTE: `xxx-yyy` is just for hiding my config here.

Proxy is configured to allow requests to https://accounts.accesscontrol.windows.net
Any suggestions on how I can better debug this situation?

The IP addresses (ETIMEDOUT 40.126.7.66:443) change almost on every request, I run. 

Also any other approach to work with PnPjs behind a corporate proxy?

### Observed Behavior
using proxy should allow PnPjs to communicate with https://accounts.accesscontrol.windows.net and all it redirects.

### Steps to Reproduce
- you should not have access to internet except through proxy.
- the proxy should be configured to allow access to https://accounts.accesscontrol.windows.net and your SPO site URL
- you may need to config npm to use the proxy and to set the proxy to allow connections to the official npm url.

It works now!

Some notes - needed to install this time @pnp/sp package too so my current dependencies are:
```
    "dependencies": {
        "@pnp/nodejs-commonjs": "^2.1.1-beta0",
        "@pnp/sp": "^2.1.1-beta0",
        "@pnp/sp-commonjs": "^2.1.1-beta0",
```
without sp it gave me:
```
> tsc

node_modules/@pnp/sp-commonjs/search/query.d.ts:36:114 - error TS2307: Cannot find module '../../common/libconfig.js' or its corresponding type declarations.

36 export declare const Search: (baseUrl: string | ISharePointQueryable, options?: IConfigOptions, runtime?: import("../../common/libconfig.js").Runtime) => ISearch;
                                                                                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@pnp/sp-commonjs/search/suggest.d.ts:10:115 - error TS2307: Cannot find module '../../common/libconfig.js' or its corresponding type declarations.

10 export declare const Suggest: (baseUrl: string | ISharePointQueryable, options?: IConfigOptions, runtime?: import("../../common/libconfig.js").Runtime) => ISuggest;
                                                                                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@pnp/sp-commonjs/utils/toabsoluteurl.d.ts:7:78 - error TS2307: Cannot find module '../../common/libconfig.js' or its corresponding type declarations.

7 export declare function toAbsoluteUrl(candidateUrl: string, runtime?: import("../../common/libconfig.js").Runtime): Promise<string>;
```

--- a/docs/sp/clientside-pages.md
+++ b/docs/sp/clientside-pages.md
@@ -298,7 +298,7 @@ part.DisplayTitle = "My New Title!";
 await page.save();
 ```
 
-> Unfortunately each webpart can be authored differently, so there isn't a way to know how the setting for a given webpart without loading it and examining how the properties are stored.
+> Unfortunately each webpart can be authored differently, so there isn't a way to know how the setting for a given webpart are stored without loading it and examining the properties.
 
 ## Page Operations
 

--- a/docs/sp/clientside-pages.md
+++ b/docs/sp/clientside-pages.md
@@ -8,7 +8,7 @@ The 'clientside-pages' module allows you to create, edit, and delete modern Shar
 | ----------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | Selective 1 | import { sp } from "@pnp/sp";<br />import { ClientsidePageFromFile, ClientsideText, ClientsideWebpartPropertyTypes, CreateClientsidePage, ClientsideWebpart, IClientsidePage } from "@pnp/sp/clientside-pages"; |
 | Selective 2 | import { sp } from "@pnp/sp";<br />import "@pnp/sp/clientside-pages";                                                                                                                                           |
-| Preset: All | import { sp, ClientsidePageFromFile, ClientsideText, ClientsideWebpartPropertyTypes, CreateClientsidePage, ClientsideWebpart, IClientsidePage } from "@pnp/sp/presets/all";                                    |
+| Preset: All | import { sp, ClientsidePageFromFile, ClientsideText, ClientsideWebpartPropertyTypes, CreateClientsidePage, ClientsideWebpart, IClientsidePage } from "@pnp/sp/presets/all";                                     |
 
 ## Create a new Page
 
@@ -260,6 +260,45 @@ page.addSection().addControl(part);
 
 await page.save();
 ```
+
+### Handle Different Webpart's Settings
+
+There are many ways that client side web parts are implemented and we can't provide handling within the library for all possibilities. This example shows how to handle a property set within the serverProcessedContent, in this case a List part's display title.
+
+```TypeScript
+import { sp } from "@pnp/sp";
+import "@pnp/sp/webs";
+import { ClientsideWebpart } from "@pnp/sp/clientside-pages";
+
+// we create a class to wrap our functionality in a reusable way
+class ListWebpart extends ClientsideWebpart {
+
+  constructor(control: ClientsideWebpart) {
+    super((<any>control).json);
+  }
+
+  // add property getter/setter for what we need, in this case "listTitle" within searchablePlainTexts
+  public get DisplayTitle(): string {
+    return this.json.webPartData?.serverProcessedContent?.searchablePlainTexts?.listTitle || "";
+  }
+
+  public set DisplayTitle(value: string) {
+    this.json.webPartData.serverProcessedContent.searchablePlainTexts.listTitle = value;
+  }
+}
+
+// now we load our page
+const page = await sp.web.loadClientsidePage("/sites/dev/SitePages/List-Web-Part.aspx");
+
+// get our part and pass it to the constructor of our wrapper class
+const part = new ListWebpart(page.sections[0].columns[0].getControl(0));
+
+part.DisplayTitle = "My New Title!";
+
+await page.save();
+```
+
+> Unfortunately each webpart can be authored differently, so there isn't a way to know how the setting for a given webpart without loading it and examining how the properties are stored.
 
 ## Page Operations
 

--- a/packages/nodejs/sptokenutils.ts
+++ b/packages/nodejs/sptokenutils.ts
@@ -1,8 +1,8 @@
 import { parse } from "url";
-import nodeFetch from "node-fetch";
 import * as jwt from "jsonwebtoken";
 import { ITypedHash } from "@pnp/common";
 import { AuthToken, SharePointServicePrincipal, ITokenCacheManager } from "./types.js";
+import { fetch } from "./net/fetch.js";
 
 class MapCacheManager implements ITokenCacheManager {
 
@@ -85,7 +85,7 @@ async function getTokenInternal(params: GetTokenInternalParams): Promise<AuthTok
     body.push(`client_secret=${encodeURIComponent(params.clientSecret)}`);
     body.push(`resource=${resource}`);
 
-    const r = await nodeFetch(params.stsUri, {
+    const r = await fetch(params.stsUri, {
         body: body.join("&"),
         headers: {
             "Content-Type": "application/x-www-form-urlencoded",

--- a/packages/sp/taxonomy/types.ts
+++ b/packages/sp/taxonomy/types.ts
@@ -124,6 +124,12 @@ export class _TermSet extends _SharePointQueryableInstance<ITermSetInfo> {
                         orderedChildren.push(found);
                     }
                 });
+                // we have a case where if a set is ordered and a term is added to that set
+                // AND the ordering information hasn't been updated the new term will not have
+                // any associated ordering information. See #1547 which reported this. So here we
+                // append any terms remaining in "terms" not in "orderedChildren" to the end of "orderedChildren"
+                orderedChildren.push(...terms.filter(info => ordering.indexOf(info.id) < 0));
+
                 return orderedChildren;
             }
             return terms;

--- a/test/nodejs/sp-extensions.ts
+++ b/test/nodejs/sp-extensions.ts
@@ -64,7 +64,7 @@ describe("nodejs - sp-extensions", () => {
                 (<any>fs).rmSync(tmpFilePath);
             } else {
                 fs.unlinkSync(tmpFilePath);
-            }            
+            }
         });
 
         it("Should allow adding chunks non-stream", async function () {

